src/scenic/simulators/carla/model.scenic

```diff
@@ -39,18 +39,36 @@ except ModuleNotFoundError:
                            'from this scenario')
 
 if 'carla_map' not in globalParameters:
-    raise RuntimeError('need to specify map before importing CARLA model '
-                       '(set the global parameter "carla_map")')
-simulator CarlaSimulator(globalParameters.carla_map)
-
-precipitation = Options({0: 70, 1: 30}) * Range(0, 100)
-param precipitation = precipitation
-param precipitation_deposits = Range(precipitation, 100)
-param cloudiness = Range(precipitation, 100)
-param wind_intensity = Range(0, 100)
-param sun_azimuth_angle = Range(0, 360)
-param sun_altitude_angle = Range(-90, 90)
-
+    param carla_map = None
+if 'address' not in globalParameters:
+    param address = '127.0.0.1'
+if 'port' not in globalParameters:
+    param port = 2000
+if 'timeout' not in globalParameters:
+    param timeout = 10
+if 'render' not in globalParameters:
+    param render = '1'
```

Comment:
Having everything be strings is indeed awkward. I could change the `--param` option to first try interpreting the given string as a number and then fall back on strings - what do you think?

Reply:
If those conditionals aren't there, the parameters defined at the scenario file might be overwritten. Is this the intended behavior or should the parameters at the scenario file have more priority?

Reply:
They shouldn't be overwritten if you define them in the scenario after the `model` statement. But I can see how this could be confusing, since the `map` and `carla_map` parameters need to be defined *before* the `model` statement, and it would be natural to have them all in one place. So maybe I should change the semantics a bit and have `param` statements at the top level always take precedence. In the mean time let's leave your code the same - thanks for the explanation.